newlib/libc/tinystdio/vfscanf.c

```diff
@@ -279,6 +279,8 @@ conv_int (FILE *stream, scanf_context_t *context, width_t width, void *addr, uin
             base = 16;
             if (!--width || IS_EOF(i = scanf_getc (stream, context)))
 		goto putval;
+            if(!isxdigit(i))
```

Comment:
There is no error here == "0xhello" should produce the value '0'. Ideally, it would also leave the input pointing at the 'x', but scanf is not required to support more than one character of look-ahead, so having it point at the 'h' instead is acceptable.